unitex/src/fr/umlv/unitex/tfst/tagging/TaggingModel.java

```diff
@@ -105,27 +150,677 @@ protected void updateModel() {
+				sequence + " isn't correct",
+				"Matching Error",
+				JOptionPane.PLAIN_MESSAGE);
+		return false;
+		
+	}
+	
+	/**
+	 * Find the sentence of the new path and return it
+	 * 
+	 * @param lst
+	 * @return the sequence of the path
+	 */
+	private String findSequence(ArrayList<Integer> lst) {
+		StringBuilder sb = new StringBuilder();
+		int tmp = 0;
+		for(Integer in : lst) {
+			
+			if(boxes[sortedNodes[in]].getBounds().getStart_in_tokens() - tmp == 2)
+				sb.append(" ");
+			tmp = boxes[sortedNodes[in]].getBounds().getEnd_in_tokens();
+			if(!getTextBoxe(boxes[sortedNodes[in]]).equals("<E>"))
+				sb.append(getTextBoxe(boxes[sortedNodes[in]]));
+		}
+		return sb.toString();
+	}
+
+
+	private boolean isKorean(char c) {
+		if((c >= 'ᄀ' &&  c <=  'ᇿ' ) ||  (c >= 'ㄱ' && c <= 'ㆎ') )
+			return true;
+		
+		return false;
+	}
+
+	private boolean isSimpleKoreanLetter(char c) {
+		String singleLetters = " ᄀ    ᆨ    ㄱ     ᄂ   ᆫ     ㄴ    ᄃ    ᆮ    ㄷ    ᄅ     ᆯ    ㄹ     ᄆ     ᆷ      ㅁ     ᄇ     ᆸ    ㅂ  ᄉ    ᆺ   ㅅ    ᄋ    ᆼ      ㅇ     ᄌ     ᆽ    ㅈ     ᄎ     ᆾ     ㅊ      ᄏ      ᆿ       ㅋ   ᄐ     ᇀ    ㅌ    ᄑ    ᇁ    ㅍ  ᄒ     ᇂ      ㅎ       ᅡ      ㅏ     ᅣ     ㅑ    ᅥ     ㅓ        ᅩ     ㅗ     ᅭ     ㅛ      ᅮ     ㅜ      ᅲ    ㅠ    ᅳ      ㅡ        ᅵ      ㅣ    ";
+		singleLetters = Normalizer.normalize(singleLetters, Normalizer.Form.NFD);
+		singleLetters = Normalizer.normalize(singleLetters, Normalizer.Form.NFKC);
+		singleLetters = Normalizer.normalize(singleLetters, Normalizer.Form.NFD);
+		return singleLetters.indexOf(c) >= 0;
+	}
+
+	private int findPreviousTokenStart(TfstGraphBox box) {
+		String txt = getTextBoxe(box);
+		if(txt.equals("<E>")) {
+			if(findPreviousBox(box).getBounds() != null) {
+				return findPreviousBox(box).getBounds().getStart_in_tokens();
+			}
+		}
+		return 0;
+	}
+	
+	private TfstGraphBox findPreviousBox(GenericGraphBox current) {
+		for (int i = 0; i < boxes.length; i++) {
+			for (GenericGraphBox next : boxes[i].transitions) {
+				if(next.equals(current)) {
+					return boxes[i];
+				}
+			}
+		}
+		
+		return null;
+	}
+
+	/** @Aissa
+	 * 
+	 * This function finds the last coordinated of each previous boxes of boxes in lstContext 
+	 * If no one is corresponding with the coordinated, we compute the value with the number
+	 * of characters
+	 * 
+	 * @param current
+	 * 			The current GenericGraphBox
+	 * 
+	 * @param lstContext
+	 * 			The list of actual contexts
+	 * 
+	 * @return the last coordinated of the current box
+	 * 			
+	 */	
+	private int findLastToken(GenericGraphBox current, ArrayList<Context> lstContext) {
+		String txt = getTextBoxe(current);
+		if(txt.equals("<E>")) {
+			if(findPreviousBox(current).getBounds() != null)
+				return findPreviousBox(current).getBounds().getEnd_in_tokens();
+		}
+		
+		for (Context context : lstContext) {
+			if(context.pos == 0 && context.currentBox > 1)
+				return boxes[sortedNodes[context.path.get(context.currentBox - 1)]].getBounds().getEnd_in_tokens();
```

Comment:
it seems this instruction returns the end-coordinate of a direct predecessor of the current box, whereas the doc tells the method returns the end-coordinate of the current box; calls to this method suggest the expected behaviour of the method is the one described in the doc